src/lib.ml

```diff
@@ -1719,3 +1719,49 @@ let to_dune_lib ({ name ; info ; _ } as lib) ~lib_modules ~foreign_objects ~dir
     ~main_module_name:(Result.ok_exn (main_module_name lib))
     ~sub_systems:(Sub_system.dump_config lib)
     ~special_builtin_support:info.special_builtin_support
+
+module Local : sig
+  type t
```

Comment:
What about `type t = private t`? A `Local.t` is basically a `t` with the invariant that `is_local t` is `true`. Then we don't need `L.to_lib`

Reply:
Sure.